java/calcite/src/main/java/org/apache/calcite/rel/externalize/MapDRelJson.java

```diff
@@ -458,7 +465,33 @@ RexNode toRex(RelInput relInput, Object o) {
         if (literal == null) {
           return rexBuilder.makeNullLiteral(sqlTypeName);
         }
-        return toRex(relInput, literal);
+
+        // omnisci serializes numeric literals differently, we need more data
```

Comment:
Do we confirm that this addition has any negative issue on the existing OmniSci engine?

Reply:
For your convinience, I tested this branch through jenkins by cherry-picking all changes into the latest master. 
And it passes all tests.